.github/workflows/build_and_test.yml

```diff
@@ -494,6 +495,43 @@ jobs:
         name: test-results-sparkr--8-${{ inputs.hadoop }}-hive2.3
         path: "**/target/test-reports/*.xml"
 
+  proto-breaking-changes-check:
+    needs: [precondition, infra-image]
```

Comment:
qq: Does the `detect` script depends on infra image (such python deps) or not?

I only saw `buf` in script and already installed in github action, so do we really need the infra-image?

Reply:
I can try without, my assumption was since this is very close to the Python build the `infra-image` is actually needed to make sure the environment is consistent.

Reply:
The build for the check has passes without the `infra-image` closing the comment.